### PR TITLE
Implemented Passport Auth

### PR DESCRIPTION
**Pull request by Blankeos**
## TL;DR
🔐 With the passport auth, you can:
- Create an account with `email`, `username`, and `password`.
- Login with either `email` or `username`.
- Logout
- Sessions are saved even if you close or refresh the browser (saved in a cookie).
- ✅ Finished APIs: `/signup` `/login` `/logout` `/currentuser`
- ⛔ Removed @burntwafflesss 's JWT implementation. (Deleted some of the APIs written so it might be hard to redo them)

## **Changes on the server**
- [x] `authRoutes.js` - Removed `/logoutall` `/profile` `/update` `/profile/remove` because I wasn't sure if they'd cause bugs to the auth. Also removed it just to clean up the files before I implemented the passport auth.
- [x] `authController.js` - Added controllers for signup, login, logout, and currentuser.
- [x] `passportConfig.js` - currently contains the local strategy for logging in with email or username. [Add other strategies](http://www.passportjs.org/docs/) here to extend passport's functionality.
- [x] `User.js` - Added `email` in User schema (but no validation).
- [x] All auth APIs have to go through `/auth/<api_name_here>`
- [x] `server.js` - cors now only listens for API requests from `http://localhost:3000` so it's secure. (Make sure to change this to the domain where the client is running in the future). The server is running on `http://localhost:4000/`.
- [x] `server.js` - Also added bodyParser  and passport middlewares on server.js
  
## **Changes on the client**
- [x] `/servers/auth.ts` - Abstracted all the axios fetches to the auth API here for easy debugging and usage. So we don't need to rewrite axios every time an auth API is needed.
- [x] `AuthTestForm.tsx` - A component that demos how auth would work in the app. Pretty much everything we need to do with authentication is implemented here so just use this as reference if you want to create a more polished Form component later on.

## Improvements that can be done
- [ ] To start using auth properly, use contexts or redux to provide a global state for the `currentUser`. This state demonstrated in `AuthTestForm.tsx`.
- [ ] OAuth - Just tweak the passportConfig.js file on the server and implement a new strategy.
- [ ] Add email validation. The signUpController is not validating if input is actually an email or not.
- [ ] Since I removed some of the functions that @burntwafflesss made, more APIs have to be redone.

---

### `/auth/signup`
Method: POST
Params: email, username, password
Purpose: Creates a new account with the given parameters, as long as they don't exist yet.

### `/auth/login`
Method: POST
Params: usernameOrEmail, password
Purpose: Logs the user in and makes the server keep track of their session.

### `/auth/logout`
Method: GET
Purpose: Logs the user out and stops keeping track of their session.

### `/auth/currentuser`
Method: GET
Purpose: Returns the user associated with the cookie that's saved in the browser. Use this on the frontend when it mounts to check if a user is currently in session.

> These APIs should only be fetched in `auth.ts` and the functions exported in `auth.ts` will be used for easy access to the APIs without typing them all over again as well as combining some functionalities
> For example, in the logIn() function in `auth.ts`, a `getCurrentUser() ` function is immediately called and returned.

### CODE REVIEW

**Review by joshxfi — COMMENTED** [review]
👏👏👏
